transcom/mymove
# INT B-20791 refactor evaluation report to a route

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20791)

## Summary

We have future work coming in upcoming Sprints that will handle GSR functionality and one of their capabilities will be adding appeals to violations and serious incidents on evaluation reports.

The way that users are reviewing evaluation reports after they are created is via modal, which we have to refactor to use a route instead since we will likely handle adding appeals to evaluation reports via modals. 

This PR does the following:
- creates the `EvaluationReportView` component which handles rendering evaluation report data, violations, serious incidents, and QAE remarks following the creation of the report
- added route to display `EvaluationReportView` component when the user clicks the `View report` button
- updated some CSS

### How to test

1. Access MM as a QAE office user
2. Using existing test moves, you can check out move code `EVLRPT`
3. Click on the `Quality Assurance` tab
4. You should now see a page with some evaluation report overviews
5. View existing reports and make sure you get no errors
6. Go and ahead and click `Create report` and make a beefy report that has SO much wrong, ensure there's a serious incident
7. Submit the report
8. Now view that report and you'll be navigated to `evaluation-report/{reportId}`
9. Check it out, review, make sure all is well and there are no errors


## Screenshots
**Old way** using a modal
![Screenshot 2024-08-12 at 11 45 28 AM](https://github.com/user-attachments/assets/e0dc3d8c-1a91-47da-bc10-efa3f2aa24af)


Click **View report** to see the refactored modal -> route
![Screenshot 2024-08-12 at 11 30 18 AM](https://github.com/user-attachments/assets/9c1e4ae5-45e5-4207-84cf-d9cf576c180b)

Out with the old modal, in with the new route
![Screenshot 2024-08-12 at 11 30 30 AM](https://github.com/user-attachments/assets/629677fd-1e0e-4780-bec6-c37ee9e5559d)

Also adding in future button functionality, but not rendering it until future work is implemented. The **Add appeal** button will be rendered for a future GSR user.
![Screenshot 2024-08-12 at 10 03 24 AM](https://github.com/user-attachments/assets/63741036-1157-4054-9a36-c15169e596bc)

508 checks are good
![Screenshot 2024-08-12 at 10 02 52 AM](https://github.com/user-attachments/assets/c29488b6-08de-4f70-bd94-2bad397fa8f2)
